File: C++/AulaOOP/Padovan.cpp
#include <stdio.h>
#define QTD_VALUES 5


using namespace std;

class Calculus{
    public:
        int val;
        int before_val = 0;
        int padovan(int x);
};

int Calculus::padovan(int x){

    switch(x){
        case 0:
            return 1;
        case 1: 
            return 1;
        case 2:
            return 1;
        default:
            return Padovan(x - 2) + Padovan(x - 3);
    }

}

bool prime(int x){

    if(x % 2 == 0 && x != 2 || x == 1){
        return false;
    }else{
        for(int i = x - 1; i > 0; i--){
            if(x % i == 0){
                return false;
            }
        }
    }
    return true;

}

int main(int argc, char *agrv[]){

    Calculus a;

    int counter = QTD_VALUES;
    for(int i = 0; counter > 0; i++){
        a.val = a.padovan(i);
        if(prime(a.val) && a.before_val != a.val){
            printf("%d", a.val);
            a.before_val = a.val;
            
        }

    }

    return 0;
}

File: Java/OOP/aulas/lista1/Padovan.java
public class Padovan {

    public static int sequence(int val){
        switch(val){
            case 0:
                return 1;
            case 1:
                return 1;
            case 2:
                return 1;
            default:
                return sequence(val - 2) + sequence(val - 3);
        }
    }

    public static boolean prime(int val){

        if(val % 2 == 0 && val != 2 || val == 1) return false;
        for(int i = val - 1; i > 1; i--){
            if(val % i == 0) return false;
        }
        return true;
    }

    public static void main(String[] args){
        int curr = 0;
        int counter = 0;
        int result;
        for(int i = 0; counter < 5; i++){
            result = sequence(i);
            if(prime(result) && curr != result){
                System.out.println(result);
                curr = result;
                counter++;
            }
        }
    }   
}